tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarymindmap.code.tex

```diff
@@ -121,6 +107,33 @@
   \pgf@process{\pgfpointtransformed{\pgfpointanchor{\tikztotarget}{west}}}%
   \advance\pgf@xa by-\pgf@x%
   \pgfkeys{/pgf/decoration/end radius/.expanded=\the\pgf@xa}%
+  % now compute lengths of connection bar flares
+  \pgfpointdiff{\pgfpointanchor{\tikztostart}{center}}%
+               {\pgfpointanchor{\tikztotarget}{center}}%
+  \pgfmathveclen@{\pgf@sys@tonumber\pgf@x}{\pgf@sys@tonumber\pgf@y}%
+  \pgf@xa=\pgfmathresult pt\relax%
+  \pgf@process{%
+    \pgfmathsetlength\pgf@x{\pgfkeysvalueof{/pgf/decoration/start radius}}%
+    \pgfmathsetlength\pgf@y{\pgfkeysvalueof{/pgf/decoration/end radius}}%
+  }%
+  \advance\pgf@x by\pgf@y\relax% x = sum of radii = 2 * nominal sum of flare lengths
+  \advance\pgf@xa by-\pgf@x\relax% xa = distance between circles = room available for flares
+  \pgf@x=.5\pgf@x\relax% x = nominal sum of flare lengths
+  \ifdim\pgf@xa>\pgf@x\relax%
+    \pgf@xa=\pgf@x\relax%
+  \fi% xa = actual sum of flare lengths
+  \edef\num{\pgf@sys@tonumber\pgf@xa}%
+  \edef\denom{\pgf@sys@tonumber\pgf@x}%
```

Comment:
This is the worst antipattern ever.  Just clobbering user-accessible names without even checking whether they are used.

Reply:
I thought the scoping rules prevented those definitions from leaking out into user scope.  If I had enclosed lines 125-127 in curly braces, would I have been correct?

Reply:
Or do I have to worry about the possibility that those names have already been allocated to registers of some special type?  Please understand that TeX is not one of my primary languages.  I seem to be encountering a lot of "gotchas".

Reply:
Which scope?  There is no enclosing scope around it which means that the scope is the entire TikZ path and someone can very well use user macros on a frontend path.